aws/resource_aws_sagemaker_endpoint_configuration.go

```diff
@@ -0,0 +1,264 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"log"
+	"time"
+)
+
+func resourceAwsSagemakerEndpointConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSagemakerEndpointConfigurationCreate,
+		Read:   resourceAwsSagemakerEndpointConfigurationRead,
+		Update: resourceAwsSagemakerEndpointConfigurationUpdate,
+		Delete: resourceAwsSagemakerEndpointConfigurationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validateSagemakerName,
+			},
+
+			"production_variants": {
+				Type:     schema.TypeSet,
```

Comment:
Is there a reason this is a TypeSet? If it's because this comes back from the api unordered then this is perfect to use. Otherwise we should use a TypeList as they are easier to access and use because of it's ordered nature.

Reply:
Done.